activerecord/test/cases/query_cache_test.rb

```diff
@@ -105,7 +111,7 @@ def test_query_cache_across_threads
 
         thread_1_connection = ActiveRecord::Base.connection
         ActiveRecord::Base.clear_active_connections!
-        assert_cache :off, thread_1_connection
+        assert_cache :empty, thread_1_connection
```

Comment:
I think this should actually assert that the connection's `query_cache_setting` has been reset back to the pool.

(Likewise, I guess we need a test verifying that given a pool with the cache enabled, if you check out a connection, disable its query cache, return the connection, then check it out again, you end up with a cache-enabled connection. Unless I'm misreading, this PR currently makes that not true.)

Reply:
I *think* we have tests that assert that, but I'm not sure they are doing it purposefully.  I'll add a specific test for it.